### PR TITLE
Add support for rendering the mouse cursor with linuxkms

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

**Review by ogoffart — COMMENTED** [review]
I feel this is a bit of a hack though

I'm wondering if this shouldn't be a feature of the Window to add an Image to the scene.
With the software renderer, we will need to make sure the previous position of the cursor is dirty.

The advantage of doing that in core is that it can be re-used for other backend that needs it (eg: uefi)
But the problem is that then the backend can't configure the mouse cursor image anymore.

**Comment by ogoffart** [comment]
> Which part exactly? The draw_image_direct bit?

I was thinking of the extra API on the renderers.

> I thought of allocating a "fake" ImageItem first, but I realise it can't be done because that also requires an ItemRc. And making that optional felt like it's too much work for this feature at this point. That's why I opted for the shorter option.

Yes true that's impossible.
Yeah, the ItemRenderer API needs change, but that's for another time.

> All in all this seemed like the best compromise to me

Ok. this is private API anyway

**Review by ogoffart — APPROVED** [review]

File: internal/backends/linuxkms/fullscreenwindowadapter.rs [hunk 1]
```diff
@@ -4,14 +4,23 @@
...
-use i_slint_core::api::PhysicalSize as PhysicalWindowSize;
+use i_slint_core::api::{LogicalPosition, PhysicalSize as PhysicalWindowSize};
+use i_slint_core::graphics::euclid;
+use i_slint_core::graphics::Image;
+use i_slint_core::item_rendering::ItemRenderer;
+use i_slint_core::slice::Slice;
+use i_slint_core::Property;
 use i_slint_core::{platform::PlatformError, window::WindowAdapter};
 
 pub trait Renderer {
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/backends/linuxkms/fullscreenwindowadapter.rs:18]: Yeah, and we have multiple `SkiaRenderer` and `FemtoVGRenderer` heh. I'll create a cleanup PR.

File: internal/backends/linuxkms/mouse-pointer.svg
Comment (ogoffart) [inline internal/backends/linuxkms/mouse-pointer.svg:1]: The cursor is barely visible on dark background.
Was there not a variant of the cursor with a white border?

Comment (tronical) [reply internal/backends/linuxkms/mouse-pointer.svg:1]: No, I was looking for that, too, but I didn't fine any in fork awesome. Please feel free to replace the svg with another one.

**Pull request merged**